File: anvil/tests/fixtures/python/bad_code/style_violations.py
"""
Python file with PEP 8 style violations.

This fixture violates multiple PEP 8 style guidelines.
"""

import json
import os  # Multiple imports on one line
import sys

# Too many blank lines


def BadFunctionName(X,y): # Function name should be lowercase, spaces after commas
    """Function with style issues."""
    z=X+y # No spaces around operator
    return z


class badClassName: # Class name should be PascalCase
    """Class with style issues."""

    def __init__(self,name,age): # Spaces after commas
        self.name=name # Spaces around =
        self.age=age

    def VeryLongMethodNameThatExceedsTheRecommendedLineLength(self, parameter1, parameter2, parameter3, parameter4, parameter5, parameter6, parameter7):
        """Method name and signature too long."""
        return parameter1+parameter2+parameter3+parameter4+parameter5+parameter6+parameter7 # Line too long


# Whitespace issues
def function_with_whitespace_issues( arg1 , arg2 , arg3 ):
    """Extra spaces in parameter list."""
    result = ( arg1+arg2 )*arg3 # Weird spacing
    return result


# Multiple statements on one line
x = 1; y = 2; z = 3


# Trailing whitespace and blank lines with whitespace
def trailing_issues():
    return "Has trailing whitespace"


# Wrong number of blank lines before class
class AnotherClass:
    pass

File: anvil/tests/fixtures/cpp/bad_code/style_violations.cpp
/**
 * C++ file with style violations.
 *
 * This fixture violates Google C++ Style Guide conventions.
 */

#include <iostream>
#include <string>
// Missing blank line before includes
#include <vector>

// Global variable (should be avoided)
int globalVariable = 42;

// Function name should be lowercase with underscores
void BadFunctionName(int X,int y) {  // No spaces after commas
    int z=X+y;  // No spaces around operators
    std::cout<<z<<std::endl;  // No spaces around <<
}

// Class name is okay, but members are not
class GoodClassName {
public:  // Should be "public:" with one space indent
    // Member variable without trailing underscore (Google style)
    int memberVar;
    std::string name;

    // Constructor with poor formatting
    GoodClassName(int x,std::string n):memberVar(x),name(n){}

    // Very long line that exceeds 80 characters and should be wrapped but is not wrapped properly
    void VeryLongMethodNameThatExceedsRecommendedLength(int param1, int param2, int param3, int param4) {
        std::cout << param1 << param2 << param3 << param4 << std::endl;
    }

private:
    void helper( int x , int y ) {  // Extra spaces in parameters
        return;
    }
};

// Multiple statements on one line
int main() { int x = 1; int y = 2; int z = x + y; return 0; }
